packages/excalidraw/components/App.tsx

```diff
@@ -2181,8 +2171,20 @@ class App extends React.Component<AppProps, AppState> {
         gridSize = this.props.gridModeEnabled ? GRID_SIZE : null;
       }
 
-      editingElement =
-        editingElement || actionResult.appState?.editingElement || null;
+      editingElement = actionResult.appState?.editingElement || null;
+
+      // make sure editingElement points to latest element reference
+      if (actionResult.elements && editingElement) {
+        actionResult.elements.forEach((element) => {
+          if (
+            editingElement?.id === element.id &&
+            editingElement !== element &&
+            isNonDeletedElement(element)
+          ) {
+            editingElement = element;
+          }
+        });
+      }
```

Comment:
Unlike previous code (introduced in https://github.com/excalidraw/excalidraw/commit/5e2f164026769fe1ab8484f200e0b6d98503ae8c#diff-86963609bec6fda3d34233676d5ae33f6aa4b9bbe2758a10cd4863b704b37093), we're updating the `editingElement` reference using the `actionResult.editingElement` instead of `this.state` which w should presume potentially stale.

In any case, it feels like that was unintentional, as we should prefer whatever the ActionResult defines.

Still, there could be regressions if the code relied on this buggy behavior. But I had to fix it to make tests pass.

/cc @ryan-di since you were recently working with `editingElement`, you may have more insight.